.github/workflows/ci.yaml

```diff
@@ -257,513 +269,532 @@ jobs:
         uses: actions/cache@v3.0.4
         with:
           path: ${{ env.PRE_COMMIT_CACHE }}
-          key: ${{ runner.os }}-${{ needs.prepare-base.outputs.pre-commit-key }}
+          key: ${{ runner.os }}-${{ needs.info.outputs.pre-commit_cache_key }}
       - name: Fail job if pre-commit cache restore failed
         if: steps.cache-precommit.outputs.cache-hit != 'true'
         run: |
           echo "Failed to restore pre-commit environment from cache"
           exit 1
       - name: Run black (fully)
-        if: needs.changes.outputs.test_full_suite == 'true'
+        if: needs.info.outputs.test_full_suite == 'true'
         run: |
           . venv/bin/activate
           pre-commit run --hook-stage manual black --all-files --show-diff-on-failure
       - name: Run black (partially)
-        if: needs.changes.outputs.test_full_suite == 'false'
-        shell: bash
-        run: |
-          . venv/bin/activate
-          shopt -s globstar
-          pre-commit run --hook-stage manual black --files {homeassistant,tests}/components/${{ needs.changes.outputs.integrations_glob }}/**/* --show-diff-on-failure
-
-  lint-flake8:
-    name: Check flake8
-    runs-on: ubuntu-latest
-    needs:
-      - changes
-      - prepare-base
-    steps:
```

Comment:
Just a tip: If I want to disable jobs for testing, I tend to add `if: false` instead of commenting all of it. That helps reduce the diff and make it more clear where the actual changes are.

Reply:
> Just a tip: If I want to disable jobs for testing, I tend to add if: false instead of commenting all of it. That helps reduce the diff and make it more clear where the actual changes are.

This is a draft and work in process, toggling comments in a code editor is easier than making small expressions hidden in existing code. The comments will be gone when this PR is ready for review; so it really doesn't matter at this point.